include/nbl/builtin/hlsl/scan/declarations.hlsl

```diff
@@ -1,66 +1,63 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
 #ifndef _NBL_HLSL_SCAN_DECLARATIONS_INCLUDED_
 #define _NBL_HLSL_SCAN_DECLARATIONS_INCLUDED_
 
 // REVIEW: Not sure if this file is needed in HLSL implementation
 
-#include "nbl/builtin/hlsl/scan/parameters_struct.hlsl"
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
 
+#ifndef NBL_BUILTIN_MAX_SCAN_LEVELS
+#define NBL_BUILTIN_MAX_SCAN_LEVELS 7
+#endif
```

Comment:
do as `NBL_CONSTEXPR` instead of define

Reply:
also this better be formulated slightly differently, in terms of reduction passes, not reduction + downsweep